internal/engine/compiler/impl_amd64.go

```diff
@@ -2772,7 +2772,7 @@ func (c *amd64Compiler) compileExtend(o *wazeroir.OperationExtend) error {
 	if o.Signed {
 		inst = amd64.MOVLQSX // = MOVSXD https://www.felixcloutier.com/x86/movsx:movsxd
 	} else {
-		inst = amd64.MOVQ
+		inst = amd64.MOVL
```

Comment:
this was a bug _if_ global i32's higher bits are set (which is also a kind of bug)